packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -753,7 +753,9 @@ export class Project extends ProjectBase {
       environment: this.renderEnvironment(props.environment, environmentVariables),
       fileSystemLocations: this.renderFileSystemLocations(),
       // lazy, because we have a setter for it in setEncryptionKey
-      encryptionKey: Lazy.stringValue({ produce: () => this._encryptionKey && this._encryptionKey.keyArn }),
+      // The 'alias/aws/s3' default is necessary because leaving the `encryptionKey` field
+      // empty will not remove existing encryptionKeys during an update (ref. t/D17810523)
```

Comment:
I'm a little worried about this change. Any we 100% certain an `undefined` key is the same as `alias/aws/s3`? Should we maybe wait for the service to fix it in their API?

How about extracting this CodeBuild change to a separate PR? It seems like we can merge the rest without this, no?

Reply:
I don't think we can merge without this.

If you have an existing pipeline for which you want to switch off the keys, all codebuild projects in them break (because they'll keep on referring to the old keys).

That seems like an extremely sharp edge I'm not too keen on rolling out because people WILL break their pipelines this way.

Reply:
Would be nice to put the explanation briefly inline here. No reason this should not be public.

Reply:
I'm confused. I added a comment right there?